src/sqlfluff/core/rules/base.py

```diff
@@ -586,13 +586,40 @@ def _log_critical_errors(error: Exception):  # pragma: no cover
         pass
 
     def _process_lint_result(
-        self, res, templated_file, ignore_mask, new_lerrs, new_fixes
+        self, res, templated_file, ignore_mask, new_lerrs, new_fixes, root
     ):
         self.discard_unsafe_fixes(res, templated_file)
         lerr = res.to_linting_error(rule=self)
         ignored = False
         if lerr:
-            if ignore_mask:
+            # Check whether this should be filtered out for being unparsable.
+            # To do that we check the parents of the anchors (of the violation
+            # and fixes) against the filter in the crawler.
+            # NOTE: We use `.passes_filter` here to do the test for unparsable
+            # to avoid duplicating code because that test is already implemented
+            # there.
+            anchors = [lerr.segment] + [fix.anchor for fix in lerr.fixes]
+            for anchor in anchors:
+                if not self.crawl_behaviour.passes_filter(anchor):  # pragma: no cover
+                    # NOTE: This clause is untested, because it's a hard to produce
+                    # edge case. The latter clause is much more likely.
+                    linter_logger.info("Fix skipped due to anchor not passing filter.")
```

Comment:
For possible future debugging purposes, do you think it may be useful to log the segment here and in the similar logic below?

Reply:
Yeah good idea.